cmake/Platform/Linux/Packaging_linux.cmake

```diff
@@ -6,53 +6,67 @@
-    # Build Libraries
-    libglu1-mesa-dev                        # For Qt (GL dependency)
-    libxcb-xinerama0                        # For Qt plugins at runtime
-    libxcb-xinput0                          # For Qt plugins at runtime
-    libfontconfig1-dev                      # For Qt plugins at runtime
-    libcurl4-openssl-dev                    # For HttpRequestor
-    # libsdl2-dev                             # for WWise/Audio
-    libxcb-xkb-dev                          # For xcb keyboard input
-    libxkbcommon-x11-dev                    # For xcb keyboard input
-    libxkbcommon-dev                        # For xcb keyboard input
-    libxcb-xfixes0-dev                      # For mouse input
-    libxcb-xinput-dev                       # For mouse input
-    zlib1g-dev
-    mesa-common-dev
-    libunwind-dev
-    pkg-config
-)
-list(JOIN package_dependencies "," CPACK_DEBIAN_PACKAGE_DEPENDS)
-
-# Post-installation and pre/post removal scripts
-configure_file("${LY_ROOT_FOLDER}/cmake/Platform/${PAL_PLATFORM_NAME}/Packaging/postinst.in"
-    "${CMAKE_BINARY_DIR}/cmake/Platform/${PAL_PLATFORM_NAME}/Packaging/postinst"
-    @ONLY
-)
-configure_file("${LY_ROOT_FOLDER}/cmake/Platform/${PAL_PLATFORM_NAME}/Packaging/prerm.in"
-    "${CMAKE_BINARY_DIR}/cmake/Platform/${PAL_PLATFORM_NAME}/Packaging/prerm"
-    @ONLY
-)
-configure_file("${LY_ROOT_FOLDER}/cmake/Platform/${PAL_PLATFORM_NAME}/Packaging/postrm.in"
-    "${CMAKE_BINARY_DIR}/cmake/Platform/${PAL_PLATFORM_NAME}/Packaging/postrm"
-    @ONLY
-)
-set(CPACK_DEBIAN_PACKAGE_CONTROL_EXTRA
-    ${CMAKE_BINARY_DIR}/cmake/Platform/Linux/Packaging/postinst 
-    ${CMAKE_BINARY_DIR}/cmake/Platform/Linux/Packaging/prerm 
-    ${CMAKE_BINARY_DIR}/cmake/Platform/Linux/Packaging/postrm
-)
+if(${PACKAGE_TYPE} STREQUAL "SNAP")
+
+    set(CPACK_GENERATOR External)
+    set(CPACK_EXTERNAL_ENABLE_STAGING YES)
+    set(CPACK_EXTERNAL_PACKAGE_SCRIPT "${LY_ROOT_FOLDER}/cmake/Platform/${PAL_PLATFORM_NAME}/Packaging_Snapcraft.cmake")
+    set(CPACK_MONOLITHIC_INSTALL 1)
+
+    set(CPACK_PACKAGING_INSTALL_PREFIX "/${CPACK_PACKAGE_NAME}/${O3DE_INSTALL_VERSION_STRING}")
+
+else()
+    #default to debian
+
+    set(CPACK_GENERATOR DEB)
+
+    set(CPACK_PACKAGING_INSTALL_PREFIX "/opt/${CPACK_PACKAGE_NAME}/${O3DE_INSTALL_VERSION_STRING}")
+
+    # get all the package dependencies, extracted from scripts\build\build_node\Platform\Linux\package-list.ubuntu-focal.txt
+    set(package_dependencies
+        libffi7
+        clang-12
+        ninja-build
+        # Build Libraries
+        libglu1-mesa-dev                        # For Qt (GL dependency)
+        libxcb-xinerama0                        # For Qt plugins at runtime
+        libxcb-xinput0                          # For Qt plugins at runtime
+        libfontconfig1-dev                      # For Qt plugins at runtime
+        libcurl4-openssl-dev                    # For HttpRequestor
+        # libsdl2-dev                             # for WWise/Audio
```

Comment:
can we somehow pull this dependency in from the Gem?

Reply:
If we do need a package as part of the list, I wonder if we can use our build node specific ones: https://github.com/o3de/o3de/blob/development/scripts/build/build_node/Platform/Linux/package-list.ubuntu-jammy.txt

Reply:
Although just to add to this, the list of libraries is unchanged from the current setup (shows as newlines due to formatting changes). I'll add a GHI to investigate possibly automating this outside of this PR.